src/clj/harja/palvelin/raportointi/raportit/vastaanottotarkastus.sql

```diff
@@ -100,3 +100,26 @@ WHERE ((:urakka::INTEGER IS NULL AND u.urakkanro IS NOT NULL)
       AND (SELECT EXTRACT(YEAR FROM lp.aika)) = :vuosi
       AND s.poistettu IS NOT TRUE
 ORDER BY s.perintapvm DESC;
+
+-- name: pkluokkien-kustannukset-hallintayksikoittain
+-- Päätellään päällystyskohteen pk-luokka laskemalla alikohteiden (yllapitokohdeosa) pituudet ja ottamalla se
+-- pk-luokka, jonka pituus on suurin
+SELECT DISTINCT ON (CONCAT(u.id, ' ', u.nimi)) CONCAT(u.id, ' ', u.nimi)          AS nimi,
+                                               o.id                               AS "hallintayksikko_id",
+                                               o.nimi                             AS "hallintayksikko_nimi",
+                                               ypk.pkluokka,
+                                               SUM(ypkk.sopimuksen_mukaiset_tyot) AS "sopimuksen-mukaiset-tyot",
+                                               SUM(ypkk.maaramuutokset)           AS "maaramuutokset",
+                                               SUM(ypkk.arvonvahennykset)         AS "arvonvahennykset",
+                                               SUM(ypkk.toteutunut_hinta)         AS "toteutunut-hinta",
+                                               SUM(ypkk.bitumi_indeksi)           AS "bitumi-indeksi",
+                                               SUM(ypkk.kaasuindeksi)             AS "kaasuindeksi",
+                                               SUM(ypkk.maku_paallysteet)         AS "maku-paallysteet"
+  FROM yllapitokohde ypk
+           LEFT JOIN yllapitokohteen_kustannukset ypkk ON ypk.id = ypkk.yllapitokohde
+           JOIN urakka u ON ypk.urakka = u.id
+           JOIN organisaatio o ON u.hallintayksikko = o.id
+ WHERE :vuosi = ANY (ypk.vuodet)
+   AND (:hallintayksikko::INTEGER IS NULL OR u.hallintayksikko = :hallintayksikko)
```

Comment:
Olisko tämä 
:hallintayksikko::INTEGER IS NULL
=> 
:hallintayksikko IS NULL 

En ei ole tietoa mitä tuo ::INTEGER cast tekee NULL arvoille, jos otetaan se vaan pois? Tämän voi kyllä katsomalla kokeilla mutta ei ole minusta ehkä normaalia käyttää integer castia null arvoon.

Reply:
Se on käsittääkseni pakko castata, jotta siitä voi varmistaa, että onko se NULL. Tämä on ainakin ollut toimintatapa Harjassa ja Napotoessa.

Reply:
Selvä sitten. Ei ollut itsellä tällaista käsitystä mutta käännän kyllä takkini jos se väärinpäin on